main/static/script.js

```diff
@@ -227,15 +198,17 @@ module.controller("mainCtrl", function(
       options.isStacked = true;
       chart.chart = new google.visualization.AreaChart(chart.dom);
     }
-    asyncRender(chart, dataTable, options);
+    $asyncRender($scope, chart, dataTable, options);
   };
+});
 
-  function receiveProfile(object, chart) {
+module.factory("$receiveProfile", function($chartWaiting, $asyncRender) {
```

Comment:
I cleaned up `$scope` and `$rootScope` everywhere, so now only the `$main`/`$embed` controllers inject `$scope` and the `$google` and `$asyncRender` factory inject `$rootScope`, and they're never passed as parameters.